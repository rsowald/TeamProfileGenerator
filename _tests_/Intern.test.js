const Intern = require('../lib/Intern');

describe('Intern class', () => {
    describe('constructor and getter methods', () => {
        const intern = new Intern('Shree Ravikumar', 625, 'ShreeRavi625@fakemail.com', 'OSU');
        expect(intern.getName()).toBe('Shree Ravikumar');
        expect(intern.getId()).toBe(625);
        expect(intern.getEmail()).toBe('ShreeRavi625@fakemail.com');
        expect(intern.getSchool()).toBe('OSU');
        expect(intern.getRole()).toBe('Intern');
    });

    it('intern card', () => {
        const intern = new Intern('Shree Ravikumar', 625, 'ShreeRavi625@fakemail.com', 'OSU');
        expect(intern.getEmployeeCard()).toMatchSnapshot();
    });
});