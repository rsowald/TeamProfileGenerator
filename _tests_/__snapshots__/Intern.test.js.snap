// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intern class intern card 1`] = `
"<div class=\\"col mb-4\\">
    <div class=\\"card h-100\\">
        <div class=\\"card-header\\">
            <h4>Intern<i class=\\"fas fa-graduation-cap\\"></i></h4>
            <h4>Shree Ravikumar</h4>
        </div>
        <div class=\\"card-body\\">
            <ul class=\\"list-group list-group-flush\\">
                <li class=\\"list-group-item\\">ID: 625</li>
                <li class=\\"list-group-item\\">Email: <a href = \\"mailto: ShreeRavi625@fakemail.com\\">getEmail() {
    return this.email;
  }</a></li>
                <li class=\\"list-group-item\\">School: OSU</li>
            </ul>
        </div>
    </div>
    </div>"
`;
